### PR TITLE
A way to turn off the scrollbar #3074

### PR DESCRIPTION
**Pull request by Puneethnaik**
issue #3074 fix. @Tyriar Please look into it. Would love to hear feedback.

### CODE REVIEW

File: src/browser/Viewport.ts [hunk 1]
```diff
@@ -131,6 +136,9 @@ export class Viewport extends Disposable implements IViewport {
       this._refresh(immediate);
       return;
     }
+    // This is for refreshing the viewport if scrollBarWidth has to be updated
+    this._refresh(immediate);
+    return;
```
Comment (Tyriar) [inline src/browser/Viewport.ts:141]: We should avoid calling `_refresh` unless we have to, could we track the last scroll bar width and only update if it's changed?

File: src/browser/Viewport.ts [hunk 2]
```diff
@@ -93,7 +93,12 @@ export class Viewport extends Disposable implements IViewport {
       this._ignoreNextScrollEvent = true;
       this._viewportElement.scrollTop = scrollTop;
     }
-
+    if (this._optionsService.getOption('scrollback') === 0) {
```
Comment (Tyriar) [inline src/browser/Viewport.ts:96]: `.options.scrollback` is better to use internally

**Review by Tyriar — APPROVED** [review]
Thanks this works pretty well, I'll just do a few clean ups before merging 👍

File: src/browser/Viewport.ts (hunk 1 above)
Comment (Puneethnaik) [reply src/browser/Viewport.ts]: Calling _refresh is costly. Therefore, we should call only when necessary; I will remember this next time. Thanks for the tip! :+1:

File: src/browser/Viewport.ts (hunk 2 above)
Comment (Puneethnaik) [reply src/browser/Viewport.ts]: Got it, Thanks for the tip! :+1:

**Pull request merged**